base/sysinfo.jl

```diff
@@ -411,7 +412,18 @@ including e.g. a WebAssembly JavaScript embedding in a web browser.
 """
 isjsvm(os::Symbol) = (os === :Emscripten)
 
-for f in (:isunix, :islinux, :isbsd, :isapple, :iswindows, :isfreebsd, :isopenbsd, :isnetbsd, :isdragonfly, :isjsvm)
+"""
+    Sys.detectwsl([os])
+
+Runtime Predicate for testing if Julia is running inside WSL.
+"""
+function detectwsl(os::Symbol)
+    islinux(os) &&
+    isfile("/proc/sys/kernel/osrelease") &&
+    contains(read("/proc/sys/kernel/osrelease", String), r"Microsoft|WSL"i)
```

Comment:
```suggestion
    contains(read("/proc/sys/kernel/osrelease", String), r"\b(Microsoft|WSL)\b"i)
```
Presumably we don't want to match if say the hypothetical "BungalowsLinux" distro had its name in here.